docs/rules/no-throw-literal.md

```diff
@@ -0,0 +1,36 @@
+# Restrict what can be thrown as an exception (no-throw-literal)
+
+This rule restrict what can be thrown as an exception.
```

Comment:
The introduction should be about the conduit of the rule. Why is it bad to throw literals? What should be done instead? Take a look at some of the existing documentation for examples.